File: array/ProblemSolvingForArrays/ArrayIntersection/intersectioArr.cpp
#include<bits/stdc++.h>
using namespace std;

    void printIntersectionArr(int arr1[],int arr2[],int m, int n){
        int i = 0,j = 0;
        while(i<m && j<n){
            if(arr1[i]<arr2[j]){
                i++;
            }else if(arr1[i]>arr2[j]){
                j++;
            }else{
                cout<<arr1[i]<< " ";
                i++;
                j++;
            }
        }
    }

int main(){

    int arr1[] = {2,4,6,7,9};
    int arr2[] = {3,4,5,6,9};

    int  m = sizeof(arr1)/sizeof(arr1[0]);
    int n = sizeof(arr2)/sizeof(arr2[0]);

    printIntersectionArr(arr1, arr2, m, n);
    return 0;
}

File: array/ProblemSolvingForArrays/ArrayIntersection/IntersectioArr.java
import java.util.*;


class IntersectioArr{

    public static ArrayList<Integer> Intersection(int arr1[],int arr2[]){
        TreeSet<Integer> set = new TreeSet<Integer>();
        for(int i : arr1){
            set.add(i);
        }
        ArrayList<Integer> list = new ArrayList<Integer>();
        for(int i: arr2){
            if(set.contains(i) && !list.contains(i)){
                list.add(i);
            }

        }
        Collections.sort(list);
        return list;
    }
    public static void main(String[] args) {
        int arr1 [] = {1,2,4,5,7};
        int arr2 [] = {2,3,4,6,7};

        ArrayList<Integer> result = Intersection(arr1, arr2);
        for(int i:result){
            System.out.println(i+ " ");
        }

    }
}